src/analysis/special_functions/integrals.lean

```diff
@@ -52,6 +53,44 @@ lemma interval_integrable_rpow {r : ℝ} (h : 0 ≤ r ∨ (0 : ℝ) ∉ [a, b])
   interval_integrable (λ x, x ^ r) μ a b :=
 (continuous_on_id.rpow_const $ λ x hx, h.symm.imp (ne_of_mem_of_not_mem hx) id).interval_integrable
 
+/-- Alternative version with a weaker hypothesis on `r`, but assuming the measure is volume. -/
+lemma interval_integrable_rpow' {r : ℝ} (h : -1 < r) :
+  interval_integrable (λ x, x ^ r) volume a b :=
+begin
+  suffices : ∀ (c : ℝ), interval_integrable (λ x, x ^ r) volume 0 c,
+  { exact interval_integrable.trans (this a).symm (this b) },
+  intro c,
+  rcases le_total 0 c with hc|hc,
+  { -- case 0 ≤ c (the interesting one)
+    have hderiv : ∀ x ∈ Ioo 0 c, has_deriv_at (λ x : ℝ, x ^ (r + 1) / (r + 1)) (x ^ r) x,
+    { intros x hx, convert (real.has_deriv_at_rpow_const (or.inl hx.1.ne')).div_const (r + 1),
+      field_simp [(by linarith : r + 1 ≠ 0)], ring, },
+    rw interval_integrable_iff_integrable_Ioc_of_le hc,
+    refine (integral_eq_sub_of_has_deriv_at_of_nonneg_of_le hderiv _ _ _ hc).1,
+    { intros x hx, apply rpow_nonneg_of_nonneg hx.1.le, },
+    { refine (continuous_on_id.rpow_const _).div_const, intros x hx, right, linarith },
+    { apply continuous_on_id.rpow_const, intros x hx, left, exact hx.1.ne', },  },
+  { -- case c ≤ 0 (of little interest, but included for completeness)
```

Comment:
I haven't tried to golf this myself, but I would think the proof you'd want would be:
  * For `x < 0`, `x ^ y = (-x)^y * cos(pi * y)`
  * Drop the `cos(pi * y)` with some `mul_const` lemma
  * Use $\int_a^b f(x) = \int_b^a f(-x)$
  * Use the "interesting direction" of the lemma.

The definitional version of $x ^ y$ for $x < 0$ is in a comment but not _quite_ how you'd want it in the library. Still, there's this, https://leanprover-community.github.io/mathlib_docs/analysis/special_functions/pow.html#complex.of_real_cpow_of_nonpos

`mul_const` doesn't seem to be in the library, though if you unfold a couple definitions it is: https://leanprover-community.github.io/mathlib_docs/measure_theory/function/l1_space.html#measure_theory.integrable.mul_const

The hardest part to me seems like it's the change of variables?